src/uct/api/tl.h

```diff
@@ -163,6 +163,12 @@ typedef struct uct_iface_ops {
                                      const uct_device_addr_t *dev_addr,
                                      const uct_ep_addr_t *ep_addr);
 
+    /* endpoint - memory registration */
+
+    ucs_status_t (*ep_mem_reg_nc)(uct_ep_h ep, const uct_iov_t *iov,
```

Comment:
not sure i would use the terminology "noncontig" since the memory _can_ be contiguous. maybe just use "ep_mem_reg"? we can decide that memory registration on an endpoint is more "generic" than over an interface
@shamisp what do you think?